docs/rules/jsx-no-undef.md

```diff
@@ -58,4 +58,8 @@ module.exports = Hello;
 
 ## When Not To Use It
 
-If you are not using JSX then you can disable this rule.
+It's recommended to avoid this rule if your project:
+
+1. does not use JSX
+2. uses TypeScript, which [automatically enables better checks than ESLint `no-undef` rules](https://typescript-eslint.io/troubleshooting/faqs/eslint/#i-get-errors-from-the-no-undef-rule-about-global-variables-not-being-defined-even-though-there-are-no-typescript-errors)
```

Comment:
item 2 and 3 are basically the same - it would be a very bad idea to turn off this rule and NOT enable the typescript-eslint rule.

Reply:
Hmm, well following the advice from the typescript-eslint docs, actually the `no-undef` rule should be disabled in TypeScript files

So option 3 is for files where TS type checking is disabled

If that doesn't make sense, I can try re-explaining

Reply:
Good question, I'm thinking that this was like this from pretty early on... can't remember a time when TS didn't give me errors for undefined JSX element identifiers 🤔